src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Utilities/Checksum.cs

```diff
@@ -44,7 +44,9 @@ public string ToBase64String()
     {
 #if NETCOREAPP
         Span<byte> bytes = stackalloc byte[HashSize];
+#pragma warning disable CS9191 // The 'ref' modifier for an argument corresponding to 'in' parameter is equivalent to 'in'. Consider using 'in' instead.
         MemoryMarshal.TryWrite(bytes, ref Unsafe.AsRef(in Data));
```

Comment:
Isn't the right fix to just change `ref` to `in` rather than disable the warning?

Reply:
Actually, I'm in this code right now. So, I'll fix it once this PR is merged.

Reply:
That would be a good fix but I think that won't work since this code multitargets several TFMs and `MemoryMarshal.TryWrite` has `ref` parameter on `net7.0` and below and `in` parameter on `net8.0`.

Reply:
Right, we have to change the `#if's`. I'll fix it. In the meantime, I think we can go ahead and merge this.